### ISSUE TITLE
Duplicate entry mysql relation attach

### ISSUE THREAD

**Issue by allanlisboaweb** [adonisjs/core#2822]
I'm having failure when trying to relate more than one ID using the attach method! When I just pass one ID everything works correctly.
  
![Captura de Tela 2021-06-21 às 22 57 31](https://user-images.githubusercontent.com/37181884/122850691-1284c680-d2e4-11eb-9ace-50d80f24ea2f.png)

<img width="1438" alt="Captura de Tela 2021-06-21 às 22 58 10" src="https://user-images.githubusercontent.com/37181884/122850747-28928700-d2e4-11eb-9ada-be45dda875da.png">

Below is my controller:

public async store({ request, response, auth }: HttpContextContract) {
    const trx = await Database.transaction()

    try {
      const validatedUserData = await request.validate({
        schema: schema.create({
          email: schema.string({}, [
            rules.unique({ table: 'users', column: 'email' }),
            rules.email()
          ]),
          password: schema.string({}, [
            rules.minLength(6)
          ])
        }),
        messages: {
          'email.required': 'E-mail obrigatório',
          'email.email': 'E-mail inválido! Informe outro e-mail e tente novamente',
          'email.unique': 'E-mail inválido! Informe outro e-mail e tente novamente',
          'password.required': 'Senha obrigatória',
          'password.minLength': 'Senha inválida. Informe no mínimo 6 caracteres'
        }
      })

      const validatedProfileData = await request.validate({
        schema: schema.create({
          name: schema.string(),
          cpf: schema.string({}, [
            rules.unique({ table: 'providers', column: 'cpf' })
          ]),
          mobile: schema.string({}, [
            rules.unique({ table: 'providers', column: 'cpf' })
          ]),
          gender: schema.enum(Object.values(Gender)),
          rating: schema.number.optional(),
          status: schema.enum.optional(Object.values(Status)),
          latitude: schema.number.optional(),
          longitude: schema.number.optional(),
          bearing: schema.number.optional(),
          notificationToken: schema.string.optional()
        }),
        messages: {
          'name.required': 'Nome obrigatório',
          'cpf.required': 'Cpf obrigatório',
          'cpf.unique': 'Cpf inválido! Informe outro Cpf e tente novamente',
          'mobile.unique': 'Telefone inválido! Por favor, tente novamente',
          'mobile.mobile': 'Telefone inválido! Por favor, tente novamente',
          'gender.required': 'Gênero inválido! Por favor, tente novamente',
          'gender.enum': 'Gênero inválido! Por favor, tente novamente'
        }
      })

      const user = await User.create({ ...validatedUserData, level: 'provider' }, { client: trx })
      user.useTransaction(trx)
      await user.related('provider').create(validatedProfileData)

      const documents = await Document.all()
      const documentsIds = documents.map(document => {
        return document.id
      })

      await user.load('provider')
      await user.provider.related('documents').attach(documentsIds)

      await trx.commit()

      const token = await auth.use('api').attempt(validatedUserData.email, validatedUserData.password)

      return { user, session: token }
    } catch (error) {
      await trx.rollback()
      console.log(error)
      return response.status(error.status).send(error.messages)
    }
  }

**Comment by thetutlage** [comment]
Can you do a `console.log(documentIds)` and share the output?

**Comment by allanlisboaweb** [comment]
Oh yeah thetutlage! Below is the print with the console log

<img width="1439" alt="Captura de Tela 2021-06-22 às 08 21 48" src="https://user-images.githubusercontent.com/37181884/122916106-e7759380-d332-11eb-8a6c-b75dda559803.png">

**Comment by thetutlage** [comment]
It looks like you have marked the `provider_id` as primary in your migration file and that means it doesn't allow duplicates

**Comment by allanlisboaweb** [comment]
Hello thetutlage! 

This table would be one of type pivo between two other tables. I found that before Create doesn't work when creating a record using the attach relationship! The "ID" field was not incremented using the UUID with beforeCreate. I solved the problem by removing the "ID" field leaving only the "PROVIDER_ID" and "DOCUMENT_ID". 

Thanks

![Captura de Tela 2021-06-23 às 10 22 00](https://user-images.githubusercontent.com/37181884/123103939-dd27c800-d40c-11eb-8233-461f3df38851.png)

![Captura de Tela 2021-06-23 às 10 27 32](https://user-images.githubusercontent.com/37181884/123104784-a43c2300-d40d-11eb-86c0-ca1912a39975.png)
